src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,40 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final Integer MIN_AGE = 18;
+    private static final Integer MIN_CHARACTERS = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RegistrationException {
+        if (user == null) {
+            throw new RegistrationException("User can't be null");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can't be null");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password can't be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("Login already exist");
+        }
+        if (user.getLogin().length() <= MIN_CHARACTERS) {
+            throw new RegistrationException("Login is too short, it should have at least: "
+                    + MIN_CHARACTERS + " characters");
+        }
+        if (user.getPassword().length() <= MIN_CHARACTERS) {
+            throw new RegistrationException("Password is too short, it should have at least: "
+                    + MIN_CHARACTERS + " characters");
+        }
```

Comment:
Similar to the login validation, the condition `user.getPassword().length() <= MIN_CHARACTERS` should be `user.getPassword().length() < MIN_CHARACTERS` to ensure that the password length is strictly greater than the minimum required characters.